README.md

```diff
@@ -10,7 +10,10 @@ Instead of manually calling `use()` for every request, `use()` is called automat
 
 ```js
 
-var superagent = require('superagent-use');
+var superagent = require('superagent');
+/* Pass the superagent object to superagent-use
+   in order to be provided with new `use` functionality */
+require('superagent-use')(superagent);
```

Comment:
I don't think it makes any difference whether we use `object-assign` or `extend` as we still need to depend on it.
About the cloning approach: other plugins don't use it but it could make sense in this case. It's not perfect though.
+ it's good because you can have many superagents with other middlewares
- it's bad because if you'd like to use the same middlewares across multiple files, then you have to either make a separate file with configuration and export (then request it instead of superagent) or repeat `use` calls in each file

Another way would be to make it optional as follows:

By default (as you suggested):
```javascript
const superagent = require('superagent'),
      superagentUse = require('superagent-use');

let withPrefix = superagentUse(superagent);
    withoutPrefix = superagentUse(superagent);

withPrefix.use(prefixMiddleware);
/* Only `withPrefix` uses the middleware. */
```

Optional (as other plugins do):
```javascript
/* config-stuff.js */

const superagent = require('superagent'),
      superagentUse = require('superagent-use');

superagentUse(superagent, { override: true });

superagent.use(prefixMiddleware);

/* other-file.js */

const request = require('superagent');
/* `request` uses the middleware */
```

If you think it's too complex, I can just go for the first option.